crates/debugging/src/tree/building/builder.rs

```diff
@@ -0,0 +1,47 @@
+use ptree::TreeBuilder;
+use ptree::item::StringItem;
+
+/// A wrapper around [`TreeBuilder`] that adds two guard nodes to the tree upon creation and removes them upon building.
+pub struct TreeBuilderWithGuard {
+    builder: TreeBuilder,
+}
+
+impl TreeBuilderWithGuard {
+    /// Creates a new [`TreeBuilderWithGuard`].
+    pub fn new() -> Self {
+        let mut builder = TreeBuilder::new("guard".to_string());
+        builder.begin_child("guard".to_string());
+        Self { builder }
+    }
```

Comment:
Could we add a short test checking that new tree has desired structure (with exactly two guards)?